pkg/yurthub/proxy/util/util.go

```diff
@@ -204,12 +211,18 @@ func WithRequestTimeout(handler http.Handler) http.Handler {
 				}
 
 				if opts.TimeoutSeconds != nil {
-					timeout := time.Duration(*opts.TimeoutSeconds+watchTimeoutMargin) * time.Second
+					var timeout time.Duration
+					if info.Verb == "watch" {
+						timeout = time.Duration(*opts.TimeoutSeconds+watchTimeoutMargin) * time.Second
+					} else if *opts.TimeoutSeconds > getAndListTimeoutReduce {
```

Comment:
please add ut testcases for get/list request

Reply:
fixed